engine/src/main/java/org/hibernate/search/spatial/impl/DistanceFilter.java

```diff
@@ -106,14 +106,16 @@ public DocIdSet getDocIdSet(LeafReaderContext context, Bits acceptDocs) throws I
 			return null;
 		}
 
+		final Bits docsWithLatitude = DocValues.getDocsWithField( atomicReader, getLatitudeField() );
+		final Bits docsWithLongitude = DocValues.getDocsWithField( atomicReader, getLongitudeField() );
 		final NumericDocValues latitudeValues = DocValues.getNumeric( atomicReader, getLatitudeField() );
 		final NumericDocValues longitudeValues = DocValues.getNumeric( atomicReader, getLongitudeField() );
 
 		return new FilteredDocIdSet( docs ) {
 			@Override
 			protected boolean match(int documentIndex) {
-				double lat = coordinate( latitudeValues, documentIndex );
-				double lon = coordinate( longitudeValues, documentIndex );
+				Double lat = coordinate( docsWithLatitude, latitudeValues, documentIndex );
+				Double lon = coordinate( docsWithLongitude, longitudeValues, documentIndex );
```

Comment:
As in the `Collector`, I think we need to keep using primitives for each lat/lon value, and you can do so by checking in advance for the fields to exist in the `Bits` instance. If not, eagerly return false.